test/powershell/Language/Scripting/ErrorHistoryId.Tests.ps1

```diff
@@ -0,0 +1,186 @@
+    }
+
+    It "Error from 'Write-Error' has the right history Id" {
+        $null = RunCommand $ps "function bar { Write-Error 'abc' }"   ## 1st in history
+        $null = RunCommand $ps "bar"                                  ## 2nd in history
+
+        $ps.HadErrors | Should -BeTrue
+        RunCommand $ps '$Error[0].InvocationInfo.HistoryId' | Should -Be 2
+    }
+
+    It "Error from 'PSCmdlet.WriteError' has the right history Id" {
+        ## 1st in history
+        $null = RunCommand $ps @'
+function bar {
+    [CmdletBinding()]
+    param()
+
+    $er = [System.Management.Automation.ErrorRecord]::new(
+        [System.ArgumentException]::new(),
+        'PSCmdlet.WriteError',
+        [System.Management.Automation.ErrorCategory]::InvalidOperation,
+        $null)
+    $PSCmdlet.WriteError($er)
+}
+'@
+        ## 2nd in history
+        $null = RunCommand $ps 'bar'
+
+        $ps.HadErrors | Should -BeTrue
+        $err = RunCommand $ps '$Error[0]'
+        $err.FullyQualifiedErrorId | Should -BeExactly "PSCmdlet.WriteError,bar"
+        $err.InvocationInfo.HistoryId | Should -Be 2
+    }
+
+    It "Error from 'PSCmdlet.ThrowTerminatingError' has the right history Id" {
+        ## 1st in history
+        $null = RunCommand $ps @'
+function bar {
+    [CmdletBinding()]
+    param()
+
+    $er = [System.Management.Automation.ErrorRecord]::new(
+        [System.ArgumentException]::new(),
+        'PSCmdlet.ThrowTerminatingError',
+        [System.Management.Automation.ErrorCategory]::InvalidOperation,
+        $null)
+    $PSCmdlet.ThrowTerminatingError($er)
+}
+'@
+        ## 2nd in history
+        $null = RunCommand $ps 'bar'
+        $ps.HadErrors | Should -BeTrue
+        $err = RunCommand $ps '$Error[0]'
+        $err.FullyQualifiedErrorId | Should -BeExactly "PSCmdlet.ThrowTerminatingError,bar"
+        $err.InvocationInfo.HistoryId | Should -Be 2
+    }
+
+    It "Error from 'Throw' has the right history Id - 1" {
+        try {
+            ## 1st in history
+            $null = RunCommand $ps '1+1'
+            ## 2nd in history
+            $null = RunCommand $ps "throw 'abc'"
+        } catch {
+            ## ignore the exception
+        }
+
+        $ps.HadErrors | Should -BeTrue
+        $err = RunCommand $ps '$Error[0]'
+        $err.FullyQualifiedErrorId | Should -BeExactly 'abc'
+        $err.InvocationInfo.HistoryId | Should -Be 2
+    }
+
+    It "Error from 'Throw' has the right history Id - 2" {
+        try {
+            ## 1st in history
+            $null = RunCommand $ps "function bar { throw 'abc' }"
+            ## 2nd in history
+            $null = RunCommand $ps 'bar'
+        } catch {
+            ## ignore the exception
+        }
+
+        $ps.HadErrors | Should -BeTrue
+        $err = RunCommand $ps '$Error[0]'
+        $err.FullyQualifiedErrorId | Should -BeExactly 'abc'
+        $err.InvocationInfo.HistoryId | Should -Be 2
+    }
+
+    It "Error from parameter binding has the right history Id" {
```

Comment:
Same here, should be turned into test cases

Reply:
Updated using `-TestCases`